make/source.mk

```diff
@@ -352,6 +346,16 @@ SIZE_OPTIMISED_SRC := $(SIZE_OPTIMISED_SRC) \
             rx/rx_bind.c \
             sensors/gyro_init.c
 
+# Gyro driver files that only contain initialization and configuration code - not runtime code
+SIZE_OPTIMISED_SRC := $(SIZE_OPTIMISED_SRC) \
+            drivers/accgyro/accgyro_mpu6050.c \
```

Comment:
Do you think it would make sense to rename these files into `..._init.c` to make it explicit that they only contain initialisation code? This will make it clear that runtime code should not go into them, and bring them in line with newer drivers like the LSM6DSO.

Reply:
I thought about that but wasn’t sure at the time whether the _init.c was going to stick.

The names might look a little weird to the modules that include the drivers with some saying init an others not, but probably not a big deal. Unless we want to leave the header named as is. Probably end up “splitting” the others at some point for consistency.

I’m open to either way.

Reply:
Not entirely sure how to best to the header file naming, but my thoughts are that the header files to be used by units that want to access the driver should contain the entire public API in one file, and not have `_init` - the consumer of the driver should not have to worry about implementation details like a split to facilitate different optimisation levels. But also open to suggestions.

Reply:
It might be overkill, but we could create basically empty “.c” files for these so every one has the pair to be consistent. These just have no runtime code at this point.

Reply:
Not sure we need to - I see it as a case where the runtime is all done in a shared file, which seems to me to be reasonable code-wise. If we create empty files this will just encourage people to put stuff into them, which in most cases will be wrong.

Reply:
@mikeller  So I looked into renaming the filenames to `..._init.c` but it's going to require updating 100+ `target.mk` legacy target files since they explicitly reference the driver files for compilation. So I propose for the sake of being able to merge this into 4.2 with low risk that we leave as it for now and follow-up with a PR to rename and update all the targets for 4.3.

Reply:
Not super worried about breaking the legacy targets, but there is also no urgency for the rename, so let's do it on two parts.

Reply:
I'm with @mikeller in that there should be one header for the entire public API and that developers can provide multiple .c files as appropriate.

I'm in favor of having `_init.c` postfix on driver initialization code as the intent of the file content is specified, and a project wide search for the `_init.c` part would (and should) reveal the compiler optimization to file selection makefile/.mk code.